### PR TITLE
Add Scalafmt

### PR DESCRIPTION
**Pull request by danicheg**
Rules are the same as in `cats`.

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version=2.7.5
```
Comment (osleonard) [inline .scalafmt.conf:1]: There is a new version we might as well go all the way to 3.0.0

Comment (danicheg) [reply .scalafmt.conf:1]: You are right, @osleonard. That work was done before 3.0.0 has been released. I'll update this PR soon.

**Comment by danicheg** [comment]
I've updated Scalafmt to 3.0.1. It would be great if you can check this PR @gemelen, @lukoyanov, @osleonard.

**Review by osleonard — APPROVED** [review]
Looks like we have a flaky tests, but looks good running the tests again to see what could be causing the issue

**Comment by lukoyanov** [comment]
LGTM.

**Review by lukoyanov — APPROVED** [review]

**Comment by danicheg** [comment]
CI workflows were failed due to #425 that will be fixed in #428.

**Comment by danicheg** [comment]
I will take another one try to green CI 🤕

File: core/src/main/scala/cats/collections/DisjointSets.scala [hunk 2]
```diff
@@ -6,17 +6,18 @@ import cats.data.State.get
...
-    *
-    * @param a Set `a`
-    * @param b Set `b`
-    * @return (new [[DisjointSets]] with updated state,
-    *   `true` if Both labels are contained and joined
-    * )
-    */
+   * Joins two disjoint sets if both are contained by this [[DisjointSets]]
+   *
+   * @param a
```
Comment (gemelen) [inline core/src/main/scala/cats/collections/DisjointSets.scala:14]: Probably worth bringing back (via inserting `-` between parts)

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
-addSbtPlugin("org.scala-js"       % "sbt-scalajs"                   % "1.7.0")
-addSbtPlugin("com.codecommit"     % "sbt-github-actions"            % "0.13.0")
+addSbtPlugin("com.47deg" % "sbt-microsites" % "1.3.4")
+addSbtPlugin("org.scalameta" % "sbt-mdoc" % "2.2.23")
+addSbtPlugin("com.eed3si9n" % "sbt-unidoc" % "0.4.3")
+addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.10.0")
+addSbtPlugin("com.github.sbt" % "sbt-release" % "1.1.0")
+addSbtPlugin("io.crashbox" % "sbt-gpg" % "0.2.1")
+addSbtPlugin("pl.project13.scala" % "sbt-jmh" % "0.4.3")
+addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
```
Comment (gemelen) [inline project/plugins.sbt:8]: This could be removed now, I guess.

**Review by osleonard — APPROVED** [review]

File: core/src/main/scala/cats/collections/DisjointSets.scala (hunk 2 above)
Comment (danicheg) [reply core/src/main/scala/cats/collections/DisjointSets.scala:14]: Scalafmt formats with a new line even with `-` 🤷🏻‍♂️

File: core/src/main/scala/cats/collections/TreeList.scala [hunk 4]
```diff
@@ -658,12 +659,13 @@ object TreeList extends TreeListInstances0 {
           else {
             val h1 = h.updated(idx, a)
             // now rebuild the front of the list
-            Trees(front.reverse_:::(h1 :: tail))
+            Trees(front reverse_::: (h1 :: tail))
```
Comment (danicheg) [inline core/src/main/scala/cats/collections/TreeList.scala:662]: To someone who interested in that fix - we've added `Scalafmt` with the `AvoidInfix` rewriting rule. So `Scalafmt` formatted `front reverse_::: (h1 :: tail)` as `front.reverse_:::(h1 :: tail)`. But that leads to behaviour changes for `reverse_:::` (see https://www.scala-lang.org/api/2.13.6/scala/collection/immutable/List.html#reverse_:::[B%3E:A](prefix:List[B]):List[B]).

**Review by gemelen — APPROVED** [review]

**Pull request merged**